### ISSUE TITLE
Suggestion: Siegewar - Add mid-siege explosions config

### ISSUE THREAD

**Issue by Goosius1** [TownyAdvanced/Towny#3786]
DESCRIPTION
* Some servers have access to plugins which allow explosions + rollbacks. In particular, cannons and tanks.
* Some have requested that that these things be more usable in sieges, by enabling explosions during sieges.
* As long as the servers make players aware that this tactic is secondary to the system  (like any sort of town infiltration), then it could be fun.
 
EXPECTED
* Add the config (Default to false)
* Read the config just like the pvp one is read, with a change for besieged towns.
* Ensure neutral towns are not affected

**Comment by suneettipirneni** [comment]
Are you suggesting support for plugins like core protect?

**Comment by Goosius1** [comment]
No, this would just be forcing on the town "Explosions" perm,  like the system already does for the town "PVP" perm.

**Comment by LlmDl** [comment]
The warzone config section has a setting for explosions as well as rollbacks. It could make sense to have siege war use teh same one as it is already used for flagwar and eventwar.

**Comment by Goosius1** [comment]
Gotcha.
I'll look at at that.

So such a shared config would have different effects depending on the war system, with different blurbs e.g

Warzone.explosions=true
# event-war : if true this will enable explosions in warzones when the war event is active.
# flag-war : ...
# siege-war : if true this will enable explosions in besieged towns.

Is this pattern ok?

**Comment by LlmDl** [comment]
I think eventually we're going to try to have a unified war settings section for things that the warmodes can share. Commenting it like that will probably be preferable as sometimes it may be different depending on war. Something as basic as explosions: true/false is pretty simple though.
